plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -4729,6 +4729,17 @@ private VirtualMachineGuestOsIdentifier translateGuestOsIdentifier(String cpuArc
         return VirtualMachineGuestOsIdentifier.OTHER_GUEST;
     }
 
+    private String translateGuestOsIdentifierEx(String cpuArchitecture, String guestOs, String cloudGuestOs) {
+        if (cloudGuestOs != null) {
```

Comment:
@bhaisaab, I did not understand what you said.

The way it is coded right now when “cloudGuestOs == null ”, it will be executed the line 4740. Do you agree with that?

My suggestion is not purely cosmetics; it is to reduce code complexity. You can maintain that behavior while removing cyclomatic complexity and that I think it is worth.